indexer/search_string_utils.cpp

```diff
@@ -466,21 +500,46 @@ UniString GetStreetNameAsKey(std::string_view name, bool ignoreStreetSynonyms)
   UniString res, suffix;
   Tokenize(name, kStreetTokensSeparator, [&](std::string_view v)
   {
-    UniString const s = NormalizeAndSimplifyString(v);
+    UniString s = NormalizeAndSimplifyString(v);
 
     if (ignoreStreetSynonyms && synonyms.FullMatch(s))
       return;
 
     if (s_directions.ApplyIf(s, [&suffix](UniString const & s) { suffix.append(s); }))
       return;
 
+    EraseDummyStreetChars(s);
     res.append(s);
   });
 
   res.append(suffix);
   return (res.empty() ? NormalizeAndSimplifyString(name) : res);
 }
 
+strings::UniString GetNormalizedStreetName(std::string_view name)
+{
+  static StreetsDirectionsHolder s_directions;
```

Comment:
Do statics here produce unnecessary multithreading overhead? Would it be better to make them a member of a class instance?